src/Nethermind/Nethermind.Hive.ConsensusWorkflowGenerator/Program.cs

```diff
@@ -0,0 +1,224 @@
+using System.Text;
+
+namespace Nethermind.Hive.ConsensusWorkflowGenerator;
+
+public static class Program
+{
+    const long MaxSizeWithoutSplitting = 35_000_000;
+    const long TargetSize = 23_000_000;
+    const long PenaltyForAdditionalInit = 300_000;
+
+    static void Main(string[] args)
+    {
+        StringBuilder fileContent = new();
+
+        List<string> directories = GetTestsDirectories();
+        Dictionary<string, long> pathsToBeTested = GetPathsToBeTested(directories);
+        List<List<string>> accumulatedJobs = GetTestsSplittedToJobs(pathsToBeTested);
+
+        WriteInitialLines(fileContent);
+
+        int jobsCreated = 0;
+        foreach (List<string> job in accumulatedJobs)
+        {
+            WriteJob(fileContent, job, ++jobsCreated);
+        }
+
+        File.WriteAllText($"{FindDirectory("Nethermind.Hive.ConsensusWorkflowGenerator")}/hive-consensus-tests.yml", fileContent.ToString());
+    }
+
+    private static Dictionary<string, long> GetPathsToBeTested(List<string> directories)
```

Comment:
Could just return lazy IEnumerable<KeyValuePair>

Reply:
In this place I can't. I need to have access to already added paths when adding new one to avoid double-testing some paths and exceeding assumed max size of single job. If there are tests with the same name in different directories e.g. `exampleTestName.json` in both `stExampleA` and `stExampleB` directories, hive `--sim.limit /exampleTestName` will test `jsons` from both directories. Because of that I want to add it only once and then if `exampleTestName` would be already present in dictionary, only sum sizes of already existed and newly added. And because of that I can't `yield return` after first occurrence